src/SimulationMap.cu

```diff
@@ -40,6 +40,15 @@ __device__ SimulationMap::SimulationMap(Polyhedron *polyhedron) :
     auto *nodes_directions = (SpacePoint *)malloc(sizeof(SpacePoint));
     nodes_directions[0] = direction_vector * mapnode_dist / get_distance(direction_vector, origin);
 
+    /*
+     * Boolean array whether the faces have nodes or not
+     * Indexing of the array is the same as indexing of `Polyhedron::faces` array
+     */
+    auto *do_faces_have_nodes = (bool *)malloc(sizeof(bool) * polyhedron->n_of_faces);
```

Comment:
I'm not sure how easy it is to understand. Please, compare your version with the following suggestion. If you think this is more complicated, mark the comment as resolved
```suggestion
    /**
     * Boolean array where i-th element tells whether the i-th face have nodes or not
     * `do_faces_have_nodes[i]` value corresponds to `polyhedron->faces[i]` face
     */
    auto *do_faces_have_nodes = (bool *)malloc(sizeof(bool) * polyhedron->n_of_faces);
```